tests/metropolis_hasting_producer_12.cc

```diff
@@ -0,0 +1,63 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2020 by the SampleFlow authors.
+//
+// This file is part of the SampleFlow library.
+//
+// The SampleFlow library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of SampleFlow.
+//
+// ---------------------------------------------------------------------
+
+
+// Test the Metropolis-Hastings producer with a diagonally elongated
+// distribution.
+
+#include <iostream>
+#include <sampleflow/producers/metropolis_hastings.h>
+#include <sampleflow/filters/conversion.h>
+#include <sampleflow/consumers/mean_value.h>
+#include <valarray>
+#include <random>
+#include <cmath>
+
+
+using SampleType = std::valarray<double>;
+
+
+double log_likelihood (const SampleType &x)
+{
+  std::valarray<double> mu = {0, 0};
+  std::valarray<double> cov = {{1, 1.5}, {1.5, 3}};
```

Comment:
This won't compile. But you can just do
```
  double mu[2] = {0,0};
  double cov[2][2] = {{1,1.5}, {1.5,3}};
```
Have you checked whether this really corresponds to positive definite matrix?